apps/analytics/analytics/processing/tile_count.py

```diff
@@ -0,0 +1,130 @@
+import time
+import pandas as pd
+
+from typing import Dict, List
+
+from analytics.mongo import RawDbClient, AnalyticsDbClient
+
+from models.analytics import TileCount
+
+pd.set_option('display.max_columns', None)
+pd.set_option('display.width', 800) 
+pd.set_option('display.max_rows', 150)  # display up to 200 rows
+
+def calculate_tile_count(
+    username: str,
+    raw_db_client: RawDbClient,
+) -> TileCount:
+    game_tick_data = raw_db_client.get_game_tick_collection().find(
+        {
+            "username": username,
+        }
+    )
+
+    ### build initial DF
+    gtdf = pd.DataFrame(game_tick_data)
+
+    gtdf['energy'] = gtdf['data'].apply(lambda row: row['energy'])
+    gtdf['sessionTickCount'] = gtdf['data'].apply(lambda row: row['sessionTickCount'])
+    gtdf['x'] = gtdf['data'].apply(lambda row: row['location']['x'] if 'location' in row and 'x' in row['location'] else None)
```

Comment:
Yeah, I think the old data had some rows without location info - but I thought at the time it may have been while the user was still logging in or something similar. Can remove ^^

Reply:
yeah if it's not needed any more then remove, otherwise alg